lib/takeaway.rb

```diff
@@ -0,0 +1,48 @@
+class Takeaway
+  attr_reader :basket
+
+  def initialize( restaurant, menu)
+    @menu = menu
+    @restaurant = restaurant
+    @basket = {}
+  end
+
+  def add( name, qty = 1)
+    fail "#{name} not available" unless @menu.contains?(name)
+    if @basket.has_key?(name) then
```

Comment:
Style/DeprecatedHashMethods: `Hash#has_key?` is deprecated in favor of `Hash#key?`. (https://github.com/bbatsov/ruby-style-guide#hash-key)<br>Style/MultilineIfThen: Do not use `then` for multi-line `if`. (https://github.com/bbatsov/ruby-style-guide#no-then)